src/farcaster/types.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Frame address type to determine if the client is requesingt  verified address or a custody address
+ *  Note: exported as public Type
+ */
+
+export enum FarcasterAddressType {
+  VerifiedAddress,
+  CustodyAddress,
+}
+
+/**
+ * GetFarcasterUserAddressRequest
+ *
+ * Note: exported as public Type
+ */
+export interface GetFarcasterUserAddressRequest {
+  fid: number;
+  farcasterAddressType: FarcasterAddressType;
```

Comment:
I am still confused where this type is coming from. Like do we ask someone to shape the data like this, or actually is coming like this from an API

Reply:
Like is `GetFarcasterUserAddressRequest` and API from somewhere?

Reply:
its not an API, its something we will ask the user to provide us

Reply:
`we ask someone to shape the data like this` -  this

Reply:
Let me know if this makes sense:
https://github.com/coinbase/onchainkit/pull/114#discussion_r1485263295